### ISSUE TITLE
[debops.nginx] Does not work with empty names list

### ISSUE THREAD

**Issue by monsterry** [debops/debops#799]
The current nginx role breaks setups in which `nginx__servers`  contains vhosts with `name` set to `[]` (configuration to use the default host).

Example:
```
nginx__servers:
  - name: []
    listen:
      - '80'
      - '[::]:80'
    location:
      '/monitoring/': |
        proxy_set_header Host              $host;
        proxy_set_header X-Real-IP         $remote_addr;
        proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Port  80;
        proxy_pass http://127.0.0.1:5000/monitoring/;
```
The error message is ` "AnsibleUndefinedVariable: list object has no element 0" `.

I narrowed the error down to line 439 in `templates/etc/nginx/sites-available/default.conf.j2`, but I am not quite sure how to fix this and take into account that `item.name` could be an empty list.

```
{% set nginx__tpl_hostname_domain = item.hostname_domain | d((item.name if item.name is string else item.name[0]).split('.')[1:] | join('.')) %}
```

**Comment by drybjed** [comment]
This should be fixed now in `master`, and will be included in the next `stable-1.0` release in a few days.

**Comment by monsterry** [comment]
Perfect bugfix. Thanks!.
